src/core/qgsexpressioncontext.cpp

```diff
@@ -630,21 +630,12 @@ QgsExpressionContextScope* QgsExpressionContextUtils::projectScope()
 
   QgsExpressionContextScope* scope = new QgsExpressionContextScope( QObject::tr( "Project" ) );
 
-  //add variables defined in project file
-  QStringList variableNames = project->readListEntry( "Variables", "/variableNames" );
-  QStringList variableValues = project->readListEntry( "Variables", "/variableValues" );
+  QgsStringMap vars = QgsProject::instance()->variables();
 
-  int varIndex = 0;
-  Q_FOREACH ( const QString& variableName, variableNames )
```

Comment:
Actually it now loops over a QMap (QgsStringMap) which I think cannot be used with Q_FOREACH.
The extra lookup comment is correct for the load project code which also is somewhere in this PR.